### PR TITLE
Add support to configure moonraker api key

### PR DESCRIPTION
**Pull request by cashew22**
Add the support to configure the moonraker api key

The PR is rebased on #20 , it should be merged after

#14

### CODE REVIEW

File: tests/test_config_flow.py [hunk 1]
```diff
@@ -51,4 +51,61 @@ async def test_tmp_failing_config_flow(hass):
...
+    assert result["errors"] == {CONF_PORT: "port_error"}
+
+
+async def test_server_api_key_config_flow(hass):
+    """Test some api key config flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    # Weird characters
```
Comment (marcolivierarsenault) [inline tests/test_config_flow.py:91]: Same thing, let's split in multiple test, easier for futur debug

File: tests/test_config_flow.py [hunk 2]
```diff
@@ -51,4 +51,61 @@ async def test_tmp_failing_config_flow(hass):
...
+
+    # Too big
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input={CONF_URL: "1.2.3.4", CONF_PORT: "4840138103"}
+    )
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["errors"] == {CONF_PORT: "port_error"}
+
+
+async def test_server_api_key_config_flow(hass):
```
Comment (marcolivierarsenault) [inline tests/test_config_flow.py:85]: We should test no/null API key

File: tests/test_config_flow.py (hunk 1 above)
Comment (cashew22) [reply tests/test_config_flow.py]: Fixed

File: tests/test_config_flow.py (hunk 2 above)
Comment (cashew22) [reply tests/test_config_flow.py]: Fixed

File: custom_components/moonraker/api.py [hunk 3]
```diff
@@ -12,10 +12,13 @@
 class MoonrakerApiClient(MoonrakerListener):
     """Moonraker communication API"""
 
-    def __init__(self, url, session, port: int = 7125):
+    def __init__(self, url, session, port: int = 7125, api_key: str = None):
         self.running = False
+        if api_key == "":
+            api_key = None
+        _LOGGER.error(api_key)
```
Comment (marcolivierarsenault) [inline custom_components/moonraker/api.py:19]: Is that for debugging purposes

**Review by marcolivierarsenault — CHANGES_REQUESTED** [review]
I think you forgot 1 debug log. 

Other than that it looks good

File: custom_components/moonraker/api.py (hunk 3 above)
Comment (cashew22) [reply custom_components/moonraker/api.py:19]: it should not be there, I forgot it

**Pull request merged**